### DISCUSSION TITLE
Does VCPKG Support CMakePresets.json?

### DISCUSSION THREAD

**Discussion by gblikas** [microsoft/vcpkg#39152] [General]
I tried tinkering around with utilizing my `CMakePresets.json` file in a port I was creating. That is, I normally build my library using the `CMakePresets.json` file, as this make different configurations, builds and installations really easy for anyone to adopt.

So, when building my port, I wanted to provide consistency for the `portfile.cmake`, via `CMakePresets.json`. I wanted to do something along the lines of

```bash
# ... 
if(VCPKG_TARGET_IS_LINUX)
  message(
    STATUS
      "VCPKG_TARGET_IS_LINUX is set. Proceeding with Linux-specific commands.")
  execute_process(COMMAND ${CMAKE_COMMAND} -S ${SOURCE_PATH} --workflow --preset
                          debug RESULT_VARIABLE RESULT)
# ... 
```

I know it's not 100% correct, as I'll need to create a way to swap workflows between debug and release, but the error I got was 
> "CMake Error: Unknown argument --workflow"

This error seems not right... Is there a tutorial on how to gracefully use CMakePresets with vcpkg `portfile.cmake`? If not, is it even supported? If it isn't supported, why not?

**Comment by dg0yt** [comment]
Portfiles are invoked via vcpkg (and additional scripts). This is script mode, not the usual project mode (CMakeLists.txt) invoked by the IDE.
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#useful-implementation-notes
And it is the triplet which controls whether to build release+debug, or only release.

HTH.

**Reply by gblikas** [reply]
@dg0yt thanks! link was very helpful.
